bot/exts/info/doc/_cog.py

```diff
@@ -0,0 +1,423 @@
+                new_symbol_name = f"{package_name}.{group_name}.{symbol_name}"
+            self.renamed_symbols[symbol_name].append(new_symbol_name)
+            return new_symbol_name
+
+    async def refresh_inventories(self) -> None:
+        """Refresh internal documentation inventories."""
+        self.refresh_event.clear()
+        await self.symbol_get_event.wait()
+        log.debug("Refreshing documentation inventory...")
+        self.inventory_scheduler.cancel_all()
+
+        self.base_urls.clear()
+        self.doc_symbols.clear()
+        self.renamed_symbols.clear()
+        await self.item_fetcher.clear()
+
+        coros = [
+            self.update_or_reschedule_inventory(
+                package["package"], package["base_url"], package["inventory_url"]
+            ) for package in await self.bot.api_client.get('bot/documentation-links')
+        ]
+        await asyncio.gather(*coros)
+        log.debug("Finished inventory refresh.")
+        self.refresh_event.set()
+
+    async def get_symbol_embed(self, symbol_name: str) -> Optional[discord.Embed]:
+        """
+        Attempt to scrape and fetch the data for the given `symbol_name`, and build an embed from its contents.
+
+        If the symbol is known, an Embed with documentation about it is returned.
+
+        First check the DocRedisCache before querying the cog's `BatchParser`.
+        """
+        log.trace(f"Building embed for symbol `{symbol_name}`")
+        if not self.refresh_event.is_set():
+            log.debug("Waiting for inventories to be refreshed before processing item.")
+            await self.refresh_event.wait()
+
+        doc_item = self.doc_symbols.get(symbol_name)
+        if doc_item is None:
+            if symbol_name.count(" "):
+                # If an invalid symbol contains a space, check if the command was invoked
+                # in the format !d <symbol> <message>
+                symbol_name = symbol_name.split(" ", maxsplit=1)[0]
+                doc_item = self.doc_symbols.get(symbol_name)
+                if doc_item is None:
+                    log.debug("Symbol does not exist.")
+                    return None
+            else:
+                log.debug("Symbol does not exist.")
+                return None
+
+        self.bot.stats.incr(f"doc_fetches.{doc_item.package}")
+
+        with self.symbol_get_event:
+            markdown = await doc_cache.get(doc_item)
+
+        if markdown is None:
+            log.debug(f"Redis cache miss for symbol `{symbol_name}`.")
+            markdown = await self.item_fetcher.get_markdown(doc_item)
+            if markdown is not None:
+                await doc_cache.set(doc_item, markdown)
+            else:
+                markdown = "Unable to parse the requested symbol."
+
+        embed = discord.Embed(
+            title=discord.utils.escape_markdown(symbol_name),
+            url=f"{doc_item.url}#{doc_item.symbol_id}",
+            description=markdown
```

Comment:
I believe I've encountered a few where the parsing algo ended up with an empty string (or all whitespace, only saw it on discord), do you propose we use `Unable to parse the requested symbol.` for the description in those cases too?

Reply:
That's not where I was going with this. I was just wondering if Discord allows empty strings as descriptions or if it wants it to be None in such cases. I'm not sure if making such substitution for empty strings is accurate. Couldn't it be a case of a symbol truly lacking a description rather than it being a result of a parsing error?

Reply:
I'm not sure if a symbol like that can exist, or at least be directly parsed with an "empty" description by the parser as it'll always try to look for something.
But functionally the description works fine with an empty string

Reply:
I'm not gonna request any changes here then, but your idea isn't totally out of the window. My say on it is to leave it as is.

Reply:
>I was just wondering if Discord allows empty strings as descriptions or if it wants it to be None in such cases.

```py
        if self.description:
            result['description'] = self.description
```
https://github.com/Rapptz/discord.py/blob/69bdc3a1849407f64ef93bd2e5119e54d6cd2208/discord/embeds.py#L589-L590

Just for reference, anything Falsey is ignored when building the payload.